test/serializers/read_attribute_for_serialization_test.rb

```diff
@@ -7,20 +7,28 @@ class ReadAttributeForSerializationTest < ActiveSupport::TestCase
       class Parent < ActiveModelSerializers::Model
         attr_accessor :id
       end
+
       class Child < Parent
         attr_accessor :name
       end
+
       class ParentSerializer < ActiveModel::Serializer
         attributes :$id
 
         define_method(:$id) do
           object.id
         end
       end
+
       class ChildSerializer < ParentSerializer
         attributes :name
       end
 
+      def test_for_serializer_initialized_with_nil_object
+        child = ChildSerializer.new(nil)
+        assert_equal nil, child.read_attribute_for_serialization(:name)
```

Comment:
> why not raise an error if object isnt an ActiveRecord

Cause we support searrializing poros. We'd have to check if the object responds to the 3 or 4 methods we require.